lib/util/smart-app-context.js

```diff
@@ -154,11 +155,17 @@ module.exports = class SmartAppContext {
 				this.locationId = data.locationId
 				this.authToken = data.authToken
 				this.refreshToken = data.refreshToken
+
+				if (!this.apiMutex) {
+					this.apiMutex = new Mutex()
+				}
+
```

Comment:
Since this seems to only be an issue for the `PAGE` lifecycle, I think it may be better to apply this fix in the constructor instead. `retrieveTokens` can be called during other lifecycles and this could potentially hide other unknown bugs down the road.